microcosm_eventsource/transitioning.py

```diff
@@ -48,7 +48,7 @@ def __call__(self, cls, state):
         return any(normalize(arg)(cls, state) for arg in self.args)
 
     def __bool__(self):
-        return any(arg for arg in self.args)
+        return all(arg for arg in self.args)
```

Comment:
This may be counterintuitive, since it ends up being the same as `AllOf`. The difference is that `nothing()` can't be an argument to `AllOf`, whereas here it can; and `nothing()` is the only element here that casts to `False`.

Reply:
I don't follow this. `AnyOf(True, False)` will now resolve to `False` instead of `True`

Reply:
@jessemyers to be honest, it's not clear to me what we're expressing with the result of `__bool__` here.
Practically speaking, the only case where we get a `False` return value here is for `Nothing()`; and we only seem to be using `__bool__` for `EventType`'s `is_initial` property:
```
@property
def is_initial(self):
    return not bool(self.value.follows)
```
so I'm interpreting the result of `__bool__` as `is_not_initial`. In that case `AnyOf(True, False)` should indeed return `False` and not `True`; and `AnyOf(*args)` should return `False` so long as there is a `False` in there, which is why I'm using `all` here.
I don't know if this generalizes, but again I have no idea what "generalizes" means here.

Reply:
OK, let me try to clarify the API here for myself and I'll get back to you.

Reply:
OK, I'm getting this now. I added 092101b with some clearer interface and unit testing. I will rebase this branch and update accordingly.

Reply:
It turns that we were both correct. Your logic was correct for `any_of`, but I was right that `any_of` and `all_of` should not be the same; `all_of` needed to be using `any()` for its boolean evaluation (and `any_of` needed to be using `all()`)